extensions/default/src/ViewerLayout/ResizablePanelsHook.tsx

```diff
@@ -264,25 +221,49 @@ const useResizablePanels = (
   }, [setRightPanelClosed]);
 
   const onRightPanelOpen = useCallback(() => {
-    const { width: panelGroupWidth } = resizablePanelGroupElemRef.current.getBoundingClientRect();
     resizableRightPanelAPIRef?.current?.expand(
-      (panelGroupDefinition.right.initialExpandedOffsetWidth / panelGroupWidth) * 100
+      getPercentageSize(panelGroupDefinition.right.initialExpandedOffsetWidth)
     );
     setRightPanelClosed(false);
   }, [setRightPanelClosed]);
 
   const onRightPanelResize = useCallback(size => {
-    if (resizableRightPanelAPIRef?.current?.isCollapsed()) {
+    if (!resizablePanelGroupElemRef?.current || resizableRightPanelAPIRef.current.isCollapsed()) {
       return;
     }
 
-    const { width: panelGroupWidth } = resizablePanelGroupElemRef.current.getBoundingClientRect();
-    const newExpandedWidth =
-      (size / 100) * panelGroupWidth - panelGroupDefinition.shared.expandedInsideBorderSize;
-
+    const newExpandedWidth = getExpandedPixelWidth(size);
     setRightPanelExpandedWidth(newExpandedWidth);
+
+    if (!isResizableHandleDraggingRef.current) {
+      // This typically gets executed when the right panel is expanded via one of the UI
+      // buttons. It is done here instead of in the onRightPanelOpen method
+      // because here we know the size of the expanded panel.
+      setMinMaxWidth(resizableRightPanelElemRef.current, newExpandedWidth);
+    }
   }, []);
 
+  /**
```

Comment:
Some nice utility functions for converting between pixels and percentages. Much more important now that we have handles of width > 0.